paymentsheet/src/test/java/com/stripe/android/customersheet/CustomerSheetActivityTest.kt

```diff
@@ -130,14 +134,60 @@ internal class CustomerSheetActivityTest {
         }
     }
 
+    @Test
+    fun `When savedPaymentMethods is not empty, payment method is visible`() {
+        runActivityScenario(
+            viewState = createSelectPaymentMethodViewState(
+                savedPaymentMethods = listOf(
+                    PaymentMethodFixtures.CARD_PAYMENT_METHOD,
+                )
+            ),
+        ) { scenario ->
+            scenario.onActivity {
+                page.waitForText("····4242")
+            }
+        }
+    }
+
+    @Test
+    fun `When showEditMenu is true, edit menu is visible`() {
+        runActivityScenario(
+            viewState = createSelectPaymentMethodViewState(
+                showEditMenu = true
+            ),
+        ) { scenario ->
+            scenario.onActivity {
+                page.waitForText("edit")
+            }
+        }
+    }
+
+    @Test
+    fun `When isGooglePayEnabled is true, google pay is visible`() {
+        runActivityScenario(
+            viewState = createSelectPaymentMethodViewState(
+                isGooglePayEnabled = true
+            ),
+        ) { scenario ->
+            scenario.onActivity {
+                page.waitForText("google pay")
+            }
+        }
+    }
+
     private fun activityScenario(
         viewState: CustomerSheetViewState,
+        providePaymentMethodName: (PaymentMethodCode) -> String,
     ): InjectableActivityScenario<CustomerSheetActivity> {
         val viewModel = mock<CustomerSheetViewModel>()
 
         whenever(viewModel.viewState).thenReturn(
             MutableStateFlow(viewState)
         )
+        whenever(viewModel.providePaymentMethodName(any())).then {
```

Comment:
I still don't like that we're mocking this :( I'll try to come up with something for this next month.